vertical-pod-autoscaler/pkg/utils/vpa/api_test.go

```diff
@@ -271,3 +271,169 @@ func TestGetContainerControlledResources(t *testing.T) {
 		})
 	}
 }
+
+func TestPatchVpaAnnotations(t *testing.T) {
+	// Create a fake VPA object to simulate the existing state
+	existingVpa := test.VerticalPodAutoscaler().WithName("test-vpa").WithNamespace("test-namespace").WithContainer(containerName).Get()
+
+	// Define the annotations to be applied
+	annotations := map[string]string{
+		"cpu_last_updated": "2024-12-10T23:57:04Z",
+		"rss_last_updated": "2024-12-10T23:57:05Z",
+	}
+
+	// Set up the fake client with the existing VPA
+	fakeClient := vpa_fake.NewSimpleClientset(existingVpa)
+
+	// Call the patchVpaAnnotations function
+	err := patchVpaAnnotations(fakeClient.AutoscalingV1().VerticalPodAutoscalers("test-namespace"), "test-vpa", annotations)
+
+	// Ensure no error occurred
+	assert.NoError(t, err, "patchVpaAnnotations should not return an error")
+
+	// Verify that a single action was performed
+	actions := fakeClient.Actions()
+	assert.Equal(t, 1, len(actions), "Expected exactly one action to be performed")
+
+	// Verify the action is a patch operation
+	action := actions[0]
+	assert.Equal(t, "patch", action.GetVerb(), "Expected the action to be a patch operation")
+}
+
+func TestAnnotationsAreStaleAndChanged(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		name                           string
+		newAnnotations                 map[string]string
+		oldAnnotations                 map[string]string
+		freshnessUpdateIntervalSeconds int
+		expectedResult                 bool
+	}{
+		{
+			name: "No changes and not stale",
+			newAnnotations: map[string]string{
+				"cpu_last_updated":      time.Now().Add(-1 * time.Hour).Format(time.RFC3339),
```

Comment:
I think this use of `time` in unit tests is fragile and might not work because consecutive invocations of `time.now()` might return different values. I think the `time.Now()` needs to be factored out to a constant and reused in unit test to make sure it is the same value.